data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py

```diff
@@ -180,6 +190,68 @@ class JoinExpression(NamedTuple):
 WHERE o.rank_order = 1
 """)
 
+ETHNICITY_QUERY_TEMPLATE = JINJA_ENV.from_string("""
+SELECT DISTINCT
+    o.* EXCEPT (rank_order)
+    {% if translate_source_concepts is not none and translate_source_concepts|length > 0 -%}{{'\t'}}   
+    REPLACE(
+        CASE {{prefix}}_source_concept_id
+        {% for translate_source_concept in translate_source_concepts %}
+            WHEN {{translate_source_concept.concept_id}} THEN {{translate_source_concept.translated_concept_id}}
+            {%- if translate_source_concept.comment is not none %} /*{{translate_source_concept.comment}}*/{%- endif %}{{'\n'}}
+        {%- endfor %}
+            ELSE {{prefix}}_concept_id
+        END AS {{prefix}}_concept_id
+    )
+    {% endif %}
+FROM 
+(
+    SELECT
+        p.person_id,
+        IF
+        (ethnicity_ob.value_as_concept_id IS NULL,
+            /*Case this out based on the race_ob (race) values, ie if it's a skip/pna respect that.*/
+            CASE race_ob.value_source_concept_id
+                WHEN  {{no_matching_concept_id}} THEN {{no_matching_concept_id}} /*missing answer*/
+                WHEN  NULL THEN {{no_matching_concept_id}} /*missing answer*/
+                WHEN  {{pna_concept_id}} THEN {{pna_concept_id}} /*PNA*/
+                WHEN  {{skip_concept_id}} THEN {{skip_concept_id}} /*Skip*/
+                WHEN  {{none_of_these_concept_id}} THEN {{none_of_these_concept_id}} /*None of these*/
```

Comment:
The `NULL` check doesn't work as expected like you mentioned, but can be fixed using:
```suggestion
            CASE 
                WHEN race_ob.value_source_concept_id = {{no_matching_concept_id}} THEN {{no_matching_concept_id}} /*missing answer*/
                WHEN race_ob.value_source_concept_id IS NULL THEN {{no_matching_concept_id}} /*missing answer*/
                WHEN race_ob.value_source_concept_id = {{pna_concept_id}} THEN {{pna_concept_id}} /*PNA*/
                WHEN race_ob.value_source_concept_id = {{skip_concept_id}} THEN {{skip_concept_id}} /*Skip*/
                WHEN race_ob.value_source_concept_id = {{none_of_these_concept_id}} THEN {{none_of_these_concept_id}} /*None of these*/
```

Do you think you can make this specific change in the RT too? We can make it in a separate ticket if you prefer, but we definitely need it before the CDR generation.

Reply:
Thanks for the reminder, adding it to RT now.

Reply:
Made the change to RT as well. I had to slightly modify the integration test.